src/app/OperationalSessionSetup.cpp

```diff
@@ -423,6 +424,50 @@ CHIP_ERROR OperationalSessionSetup::LookupPeerAddress()
     return Resolver::Instance().LookupNode(request, mAddressLookupHandle);
 }
 
+void OperationalSessionSetup::PerformLookupOnExistingSession()
+{
+    bool isConnected = false;
+
+    if (mPerformingLookupOnConnectedSession)
+    {
+        // We are already in the middle of a lookup from a previous call to
+        // PerformLookupOnExistingSession. In that case we will just exit right away as
+        // we are already looking to update the results from the previous lookup.
+        return;
+    }
+
+    if (mState != State::NeedsAddress)
+    {
+        // The expectation is if we are not already performing and address lookup on an existing
+        // session that we should be recently allocated which implies we should be in the
+        // State::NeedsAddress state. Something has gone pretty wrong so we call
+        // DequeueConnectionCallbacks which will release ourself.
+        VerifyOrDie(false);
+        DequeueConnectionCallbacks(CHIP_ERROR_INCORRECT_STATE);
+        // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
+        return;
+    }
+
+    isConnected = AttachToExistingSecureSession();
```

Comment:
Why do we need this bit?  I don't think we want it at all...  It will fail for the defunct session case, right?

Reply:
Isn't that what we want to do? If the session is defunct then we don't want to do an address lookup right?

Reply:
No, I think we do...

Defunct means we should not try to do new exchanges on the session and set up a new one instead.  But for existing exchanges or interactions that are already tied to that session, we should try to see whether we can get through, and that means updating the address.

Put another way, if DHCP goes through and changes IP addresses, we try to send on our session and the first send fails, and at the same time we happen to not get an expected subscription report (because _our_ address changed too), we don't want to just give up on our send without re-resolving.